independent-projects/bootstrap/app-model/src/main/java/io/quarkus/bootstrap/workspace/SourceDir.java

```diff
@@ -22,6 +26,8 @@ default boolean isOutputAvailable() {
 
     Path getOutputDir();
 
+    Path getGeneratedSourcesDir();
```

Comment:
So far, this interface was meant to be mapping one src dir to an output dir.
You could represent the generated sources dir with a dedicated instance of `SourceDir` mapped to the same output dir as the main source dir. And then `DefaultArtifactSource` accepts collection sources and resources.

Reply:
> You could represent the generated sources dir with a dedicated instance of SourceDir

With a boolean property `isGenerated`? This is used as output, I need to find it back when calling the compiler.

Reply:
It seems like somehow it has to be annotation processor-specific, given that there are other generated sources that won't be relevant for the rest of this change. `isAtpGenerated()`?

Reply:
Conceptually, I think this is not just another source dir.

First, because Quarkus does not have to look at it: it doesn't have to track it, it doesn't have to compile stuff that is in there.

Second, because it's actually both an output of a given source folder (APT outputs sources in there based on a regular Maven/Gradle source folder), and an input that produces bytecode to the same output folder as the specified source folder, except that input is automatically handled by javac and we don't have to look at it or invoke javac on it.

IMO, it's an intermediary by-product of the regular compilation.

If I add a dedicated `SourceFolder` then every plase we handle them, we'll have to skip them with `@isAptGenerated()` which seems like a lot of error-prone work, when really we're only using this location info to pass it from Maven (or Gradle) to javac.

I mean, if you insist, I can do this change, and you know this code/arch better, for sure, but I feel like this is not the right way to attach this info.

Reply:
OK, done, and rebased.